src/v/compat/init_tm_tx_compat.h

```diff
@@ -32,21 +32,27 @@ struct compat_check<cluster::init_tm_tx_request> {
       cluster::init_tm_tx_request obj, json::Writer<json::StringBuffer>& wr) {
         json_write(tx_id);
         json_write(transaction_timeout_ms);
+        json_write(expected_pid);
         json_write(timeout);
     }
     static cluster::init_tm_tx_request from_json(json::Value& rd) {
         cluster::init_tm_tx_request obj;
         json_read(tx_id);
         json_read(transaction_timeout_ms);
+        json_read(expected_pid);
         json_read(timeout);
         return obj;
     }
+
+    // New redpanda version will use only serde for init_tm_tx request, and
+    // expected_pid can be handled only this version. So we do not need to
```

Comment:
it looks like this request existed in previous release, so how is it safe to use serde-only if during a rolling upgrade an old/new node are communicating with each other?

Reply:
How it works in my opinion. 
expected pid should be included only in this new redpanda release (22.3). So old nodes should not get it in internal rpc. So I only put it in serde, because adl version uses only old redpanda versions which do not need expected_pid